frontend/src/lib/lemon-ui/VerticalNestedDND/VerticalNestedDND.tsx

```diff
@@ -0,0 +1,845 @@
+import {
+    closestCenter,
+    CollisionDetection,
+    defaultDropAnimationSideEffects,
+    DndContext,
+    DraggableSyntheticListeners,
+    DragOverlay,
+    DropAnimation,
+    getFirstCollision,
+    MeasuringStrategy,
+    MouseSensor,
+    pointerWithin,
+    rectIntersection,
+    TouchSensor,
+    useSensor,
+    useSensors,
+} from '@dnd-kit/core'
```

Comment:
I used dnd kit for this as we already use it as a dependency elsewhere. Most of this file is modified examples that they provide